### PR TITLE
[Feature](mluOpSyncBatchNormBackwardReduce): Add new API and deprecate old ones

### PR DESCRIPTION
**Pull request by duzekunKTH**
Thanks for your contribution and we appreciate it a lot. :rocket::rocket:

## 1. Motivation

算子api格式错误，新开api并废弃旧的。

## 2. Modification

	modified:   kernels/sync_batchnorm_backward_reduce/sync_batchnorm_backward_reduce.cpp
	modified:   mlu_op.h

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS™ Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- static threshold
  - diff1
    - [ ] float32 mlu diff1 <= 1e-5
    - [ ] float32 mlu diff1 <= 3e-3
    - [ ] float16 mlu diff1 <= 3e-3
  - diff2
    - [ ] float32 mlu diff2 <= 1e-5
    - [ ] float32 mlu diff2 <= 3e-3
    - [ ] float16 mlu diff2 <= 3e-3
  - diff3
    - [ ] mlu diff3 == 0
    - [ ] mlu diff3_1 == 0
    - [ ] mlu diff3_2 == 0
- dynamic threshold
  - [ ] diff1: mlu diff1 <= max(baseline diff1 * 10, static threshold)
  - [ ] diff2: mlu diff2 <= max(baseline diff2 * 10, static threshold)
  - [ ] diff3: mlu diff3 <= max(baseline diff3 * 10, static threshold)
    - float32, threshold = 1e-5
    - float16, threshold = 1e-3

#### 3.1.2 Operator Scheme checklist

- Supported hardware
  - [x] MLU370
  - [x] MLU590
- Job types
  - [ ] BLOCK
  - [ ] UNION1
  - [ ] UNION2
  - [ ] UNION4
  - [ ] The operator will dynamically select the most suitable task type, for example, UNION8

#### 3.2.1 Accuracy Test

[^      OK ] /workspace/volume/mluops-develop/SOFT_TRAIN/release_test/sync_batchnorm_backward_reduce/sync_batchnorm_backward_reduce_9_manual_float32_float32_1633658583365.pb
[       OK ] sync_batchnorm_backward_reduce/TestSuite.mluOp/37 (3 ms)
[----------] 38 tests from sync_batchnorm_backward_reduce/TestSuite (1760 ms total)

[----------] Global test environment tear-down
[ SUMMARY  ] Total 38 cases of 1 op(s).
ALL PASSED.
[==========] 40 test cases from 3 test suites ran. (6482 ms total)
[  PASSED  ] 40 test cases.

- [ ] Data type test (e.g. float32/int8)
- [ ] Multi-dimensional tensor test
- [ ] Layout test
- [ ] Different size/integer remainder end segment/alignment misalignment test
- [ ] Zero dimensional tensor test/zero element test
- [ ] stability test
- [ ] Multiple platform test
- [ ] Gen_case module test, see: [Gencase-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/Gencase-User-Guide-zh.md)
- [ ] Nan/INF tests 
- [ ] Bug fix tests
- [ ] For memory leak check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md)
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md)
- [ ] For I/O calculation efficiency check details, see: [MLU-OPS™-Performance-Acceptance-Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md)

#### 3.2.2 Parameter Check

Test Point-1: `When a new operator is submitted, the test points are given and the test results are stated`. Acceptance Standard: `Normal error`.
```bash
Please fill your test results(Error Message) in here, ...
```

Test Point-2: `Whether illegal parameters are passed`. Acceptance Standard: `Normal error`.
```bash
Test results...
```

### 3.3 Performance Test

See [MLU-OPS™ Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

**Review by DanieeelLiu — APPROVED** [review]

File: kernels/sync_batchnorm_backward_reduce/sync_batchnorm_backward_reduce.cpp [hunk 1]
```diff
@@ -44,7 +44,14 @@ mluOpStatus_t MLUOP_WIN_API mluOpGetSyncBatchnormBackwardReduceWorkspaceSize(
   return MLUOP_STATUS_SUCCESS;
 }
 
-mluOpStatus_t MLUOP_WIN_API mluOpSyncBatchnormBackwardReduce(
+mluOpStatus_t MLUOP_WIN_API mluOpGetSyncBatchnormBackwardReduceWorkspaceSize(
+    mluOpHandle_t handle, const mluOpTensorDescriptor_t desc_x,
+    size_t *workspace_size) {
+  return mluOpGetSyncBatchNormBackwardReduceWorkspaceSize(
```
Comment (PetrelYy) [inline kernels/sync_batchnorm_backward_reduce/sync_batchnorm_backward_reduce.cpp:50]: = warning log

File: mlu_op.h [hunk 2]
```diff
@@ -7663,6 +7663,10 @@ mluOpGetRoiAwarePool3dForwardWorkspaceSize(mluOpHandle_t handle,
  * also performs max pooling or average pooling on the voxels and results in \b argmax
  * and \b pooled_features.
  *
+ * @par Deprecated
+ * - :: mluOpRoiawarePool3dForward is deprecated and will be removed in the future
```
Comment (PetrelYy) [inline mlu_op.h:7667]: :: mluOpRoiawarePool3dForward  ，有空格，文档打包会失败

File: kernels/sync_batchnorm_backward_reduce/sync_batchnorm_backward_reduce.cpp (hunk 1 above)
Comment (duzekunKTH) [reply kernels/sync_batchnorm_backward_reduce/sync_batchnorm_backward_reduce.cpp:56]: 已修改

File: mlu_op.h (hunk 2 above)
Comment (duzekunKTH) [reply mlu_op.h]: 已修改

**Review by PetrelYy — APPROVED** [review]

**Review by chqy99 — APPROVED** [review]

**Pull request merged**